service/debugger/debugger.go

```diff
@@ -1467,6 +1467,65 @@ func (d *Debugger) Functions(filter string) ([]string, error) {
 	return funcs, nil
 }
 
+func traverse(t proc.ValidTargets, f *proc.Function, depth int, FollowCalls int) ([]string, error) {
```

Comment:
You should keep track of which functions have been visited already and avoid revisiting them.

Reply:
Sure, but you can't set more than one breakpoint on D.

Reply:
After some discussion with @derekparker we realized that we may end up going further down the call graph of the function depending on how we got to it and how far down in depth we are... so there can be a situation we may miss on a child if we short circuit the traversal  
alternatively looking at it, we can come to a function from two different parents and we would then need to traverse such a function twice anyway

Reply:
> After some discussion with @derekparker we realized that we may end up going further down the call graph of the function depending on how we got to it and how far down in depth we are

True, let's make this a breadth first visit instead.

Reply:
Just to stress why this is important: tracing a function that calls fmt.Printf with `--follow-calls=10` on my system takes 10 seconds, just to set up the breakpoints. The number of breakpoints it ends up setting is relatively small (222) but to do that it ends up disassembly the same functions thousands of times. For example `fmt.(*buffer).writeString` is visited by `traverse` 4727 times (and redisassembled each time), `runtime.panicIndex` is visited 7900 times.

This implementation is extremely inefficient.